### ISSUE TITLE
Docs Suggestion: Example section introductions

### ISSUE THREAD

**Issue by tylerflex** [flexcompute/tidy3d#1446]
Kind of a minor comment but I think it will improve the docs a bit.

For each of our docs sections, there is a page with a bunch of links to the specific notebooks. For example, in the "design space exploration" section, it looks like this

https://docs.flexcompute.com/projects/tidy3d/en/v2.6.0rc1/notebooks/docs/features/parameter_sweep.html

I think what would improve things a bit is to put a few paragraphs of introduction on this page, explaining what these examples are showing and which ones might be best to check out depending on what you want to do. For example, this page could include some text above the example links like:

```
Tidy3D offers a few tools that facilitate and automate the exploration of a user-defined design space. The `design` plugin is a wrapper that allows users to define a general "parameter space" and sample this space through tidy3d simulations. The result is returned to the user in a `pandas.DataFrame` format for easy post processing and analysis. For a detailed tutorial, see [Design Plugin Tutorial Link].  We also have tools for batch processing of simulations, which can make custom parameter sweeps convenient. For an example of batching simulations and other tools for managing the results in a more custom way, see the [Parameter Sweep Tutorial Link] example.
```

**Comment by tomflexcompute** [comment]
Yeah sounds like a great idea. Can prepare the paragraphs for each section and then @daquinteroflex can incoperate them into the page

**Comment by tylerflex** [comment]
I think they can be written into these [pages](https://github.com/flexcompute/tidy3d-notebooks/blob/pre/2.6/docs/features/parameter_sweep.rst), but dario can comment.

**Comment by daquinteroflex** [comment]
Hmm let me have a think how best to do this as it might require a custom toctree, but we can go preparing the paragrahs anyway for sure (maybe this will solve the multiple style header issue you both suggested too)

**Comment by tomflexcompute** [comment]
I drafted some intro paragraphs for the case studies. @tylerflex @daquinteroflex feel free to take a look [here](https://www.notion.so/flexcompute/Section-introduction-paragraphs-4dc69c3856614da19085751303861a00) and provide any edits as you see fit. After you think the texts look good, we can implement it and see how it looks. Then we can do the same for the Python tutorial sections.

**Comment by tylerflex** [comment]
Thanks @tomflexcompute  these look really good. Would it be useful for me to do a few? such as the design space exploration and adjoint? 

As a micro-optimization: I think in general the more links we can put throughout the docs, the better, so whenever we reference an example or other section we can add a link, but just the text for now will give us a big improvement I think.

**Comment by tomflexcompute** [comment]
Yeah absolutely. If you can write a few, please add them to the notion page. I'll continue adding paragraphs for the python tutorial sections as well in the upcoming week. 

For the adjoint examples, I think we briefly thought about moving them to the case study and creating a new category. Not sure if we want to have them in the Photonic Design Optimization section or have a separate section like Adjoint Optimization and change the current Photonic Design Optimization to something like Gradient-free Optimization?

**Comment by tylerflex** [comment]
I added a few sections.

I'd say we could either have (1)
A) Optimization (adjoint + global optimization)
B) Design Space Exploration (sweep/batch)

or (2) just separate inverse design out:
A) Inverse design (adjoint only)
B) Design Space Exploration (sweep/batch + global optimization)

Which do you think makes more sense? I'm maybe more in favor of the 2nd but only slightly, only because we have almost 15 adjoint examples so far.

**Comment by tomflexcompute** [comment]
Thanks Tyler. I think either way works. The first approach is slightly more convenient as we only need to move the adjoint examples from tutorials to case studies. In approach 2, we will move the sweep and batch notebooks to case studies? They are more like tutorials than case studies but could be fine as well.

**Comment by tylerflex** [comment]
I'm not really sure, maybe I'll leave it up to you to decide. I dont have any strong preferences.

Side note: we should consider I think renaming the "FDTD Adjoint Optimization" header to maybe "Inverse Design Optimization" or something. I dont know if people know Adjoint optimization?

**Comment by tomflexcompute** [comment]
Just realized that this fell through the cracks and I never end up finishing it. It's close to finish though. I wonder if @FilipeFcp can help wrap it up?

**Comment by daquinteroflex** [comment]
Whoops sorry this slipped me too with the other docs issues. 

Sure, that sounds great!

@FilipeFcp would you be interested in learning sphinx? I have an implementation approach that I think should work for this - if you're interested we can work together on it. The idea would be to implement a `custom-toctree` or similar command that would wrap the `toctree` operator alongside a description which would render in the documentation. It'd follow a standard[ sphinx command structure](https://www.sphinx-doc.org/en/master/development/tutorials/extending_build.html#using-the-extension).

Still debating whether this needs to be within our custom sphinx-extension or whether it's easier to hack it onto our `conf.py`, but because it's an endeavour have been focusing on the other things I've been working on. In the long run, I think it'd benefit everyone to have our own documentation extension and I began [one](https://github.com/daquintero/sphinxcontrib-autoflex) when I did my internship. We can meet if you like to plan this out.

**Comment by FilipeFcp** [comment]
Hi @daquinteroflex. That sounds interesting! I’d be interested in learning Sphinx and working on this with you. I’ll take a look at the documentation to get started, then we can set up a meeting.

**Comment by daquinteroflex** [comment]
Will transfer this to our custom `autoflex` documentation extension exploration project
